src/log_unstable.rs

```diff
@@ -74,22 +74,25 @@ impl Unstable {
     // is any.
     pub fn maybe_term(&self, idx: u64) -> Option<u64> {
         if idx < self.offset {
-            if self.snapshot.is_none() {
-                return None;
+            match self.snapshot.as_ref() {
```

Comment:
Seem we used a Go style here before, and maybe this is a better Rust style.

PTAL @BusyJay

Reply:
I think less indent is better is a common sense. Besides using `?` is better here.

Reply:
Using `?` here aggravates the borrow checker.

Reply:
aggravate? you mean it will reduce the performance?

Reply:
It prevents compiling.

Reply:
Seems fine: https://play.rust-lang.org/?gist=000c63feeb789556fe1c0e1b17186184&version=stable

Reply:
any update @Hoverbear

Reply:
Didn't see this, reflected.